src/plugins/workspace/server/utils.ts

```diff
@@ -113,3 +115,23 @@ export const getDataSourcesList = (client: SavedObjectsClientContract, workspace
       }
     });
 };
+
+export const checkAndSetDefaultDataSources = async (
```

Comment:
```suggestion
export const checkAndSetDefaultDataSource = async (
```
defaultDataSource should be singular I suppose.

And I guess there are some similar logic in DSM that sets default data source, can we try reuse that?